### PR TITLE
fix: Remove erroneous <br> tags from error messages

### LINKED ISSUE #28452: Spacing of error message is too long with <br>
### Current behavior

image (4)
image (5)

The error messages appear to have really wide spacing. They didn't used to have this. Maybe it changed in 10.0. 

### Desired behavior

Spacing should not be so far apart in error messages. We should look at older versions of Cypress and how it used to look to compare when it changed and reduce spacing. Will have to update a lot of snapshots.

### Test code to reproduce

cy.click on non-visible element

### Cypress Version

13.6.0

### Node version

18

### Operating System

Mac

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by jennifer-shehane**
- Closes https://github.com/cypress-io/cypress/issues/28452

### Additional details

There was a regression introduced in 12.4.0 with [this PR](https://github.com/cypress-io/cypress/pull/25467), where all of our error messages were formatting to include `<br>` tags when there was a new line in the error message.

The original PR did not run Percy checks since it was an outside contributor, so this visual change went unnoticed.

This adds an exception to the formatting logic when the message is of type: error in order to maintain the functionality of the originally intended PR to have line breaks in Cypress.log messages.

### Steps to test

A test was added to cover this in the test_errors.cy.ts, we should also see Percy snapshots update.

#### Before

image (4)
image (5)

#### After

Screen Shot 2023-12-11 at 11 22 18 AM

Screen Shot 2023-12-11 at 12 12 39 PM

### PR Tasks

- [x] Have tests been added/updated?
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by brian-mann** [comment]
Can you add a test for the affirmative case where it **does** keep `<br />` for `Cypress.log(...)` ? If they already exist, then just point that out.

**Comment by jennifer-shehane** [comment]
@brian-mann Yah these original tests are still there: https://github.com/cypress-io/cypress/pull/25467/files#diff-95c9055c314a4b42dab5687daf7ea43804e5ec4d3d613632153b0dbde2d33e96R56

File: packages/reporter/src/commands/command.tsx [hunk 1]
```diff
@@ -24,12 +24,14 @@ import HiddenIcon from '@packages/frontend-shared/src/assets/icons/general-eye-c
...
-const md = new Markdown({ breaks: true })
-
 const displayName = (model: CommandModel) => model.displayName || model.name
 const nameClassName = (name: string) => name.replace(/(\s+)/g, '-')
 
-export const formattedMessage = (message: string) => {
+export const formattedMessage = (message: string, type: string) => {
+  // we don't want <br> in our error messages, but allow it in Cypress.log
+  const mdObj = type === 'error' ? {} : { breaks: true }
+  const md = new Markdown(mdObj)
```
Comment (chrisbreiding) [inline packages/reporter/src/commands/command.tsx]: I worry there's a performance cost to instantiating this every time we format a message. An alternative would be to instantiate one with `breaks: true` and one without (above, once) and then choose which one to use when formatting messages here.

Comment (jennifer-shehane) [reply packages/reporter/src/commands/command.tsx]: True story. I'll refactor this.

Comment (jennifer-shehane) [reply packages/reporter/src/commands/command.tsx]: https://github.com/cypress-io/cypress/pull/28496/commits/a2c8ca322523d858a5ba69fe69d2a89e4ad58871

**Review by chrisbreiding — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**